From 9299de110ec4e03c1cdfeb357f54d4a536aedea5 Mon Sep 17 00:00:00 2001
From: Arulselvan M <arulselvan.m@intel.com>
Date: Thu, 30 May 2013 16:42:34 +0530
Subject: [PORT FROM MAIN]Ending time of RTSP streaming to show ""-.--.--" instead of "0:00".

BZ: 35440

Setting -1 for rtsp streaming contents which are unable to provide valid duration
information.

Change-Id: I2db550d6dc7c35800bd116952af60faf1f7dbaaa
Orig-Change-Id: I36c8a9b3f727ad289dba3264203fb52f97906d11
Signed-off-by: Arulselvan M <arulselvan.m@intel.com>
---
 .../libmediaplayerservice/nuplayer/RTSPSource.cpp  |    2 +-
 media/libstagefright/rtsp/APacketSource.cpp        |    2 +-
 2 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/media/libmediaplayerservice/nuplayer/RTSPSource.cpp b/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
index 3958b97..51075ec 100644
--- a/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
+++ b/media/libmediaplayerservice/nuplayer/RTSPSource.cpp
@@ -280,7 +280,7 @@ void NuPlayer::RTSPSource::setEOSTimeout(bool audio, int64_t timeout) {
 }
 
 status_t NuPlayer::RTSPSource::getDuration(int64_t *durationUs) {
-    *durationUs = 0ll;
+    *durationUs = -1ll;
 
     int64_t audioDurationUs;
     if (mAudioTrack != NULL
diff --git a/media/libstagefright/rtsp/APacketSource.cpp b/media/libstagefright/rtsp/APacketSource.cpp
index 0ebf994..952866a 100644
--- a/media/libstagefright/rtsp/APacketSource.cpp
+++ b/media/libstagefright/rtsp/APacketSource.cpp
@@ -420,7 +420,7 @@ APacketSource::APacketSource(
     if (sessionDesc->getDurationUs(&durationUs)) {
         mFormat->setInt64(kKeyDuration, durationUs);
     } else {
-        mFormat->setInt64(kKeyDuration, 60 * 60 * 1000000ll);
+        mFormat->setInt64(kKeyDuration, -1);
     }
 
     mInitCheck = OK;
-- 
1.7.4.1

